### DISCUSSION TITLE
Is it possible to backtest a pairs trade?

### DISCUSSION THREAD

**Discussion by tankorsmash** [kernc/backtesting.py#317] [Q&A]
I'd like to short one symbol and go long another at the same time. 

I've got all the OLHC data for both combined into one dataframe as separate columns, but it doesn't seem like the library supports going long and shorting at the same time, since they're different prices/columns during the same bar/row.

`backtesting.py`'s `last_price()` uses `self.data._Close` so it doesn't look like it supports any sort of customization there. 

A solution that seems I could try on my own is to run two backtests and then combine their results somehow.

**Comment by kernc** [comment]
Shorting an instrument is like going long its _inverse_. Therefore, if you have BTCUSD and ETHUSD and want to buy BTC and short ETH, let's say, you need to _buy_ a derived instrument:
```text
X = BTCUSD/ETHUSD
```
You can pass this `X` as the backtest `data=` to get the library to compute P&L correctly (don't forget to double the commission), and in it you can also pass extra columns (BTCUSD's and ETHUSD's prices) to base your signals on.

Alternatively, you could run two backtests, sync their signals, and afterwards combine their result `stats._trades` by something like https://github.com/kernc/backtesting.py/pull/281.
